pkg/util/html/form.go

```diff
@@ -0,0 +1,115 @@
+package html
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"golang.org/x/net/html"
+)
+
+func visit(n *html.Node, visitor func(*html.Node)) {
+	visitor(n)
+	for c := n.FirstChild; c != nil; c = c.NextSibling {
+		visit(c, visitor)
+	}
+}
+
+func GetElementsByTagName(root *html.Node, tagName string) []*html.Node {
+	elements := []*html.Node{}
+	visit(root, func(n *html.Node) {
+		if n.Type == html.ElementNode && n.Data == tagName {
+			elements = append(elements, n)
+		}
+	})
+	return elements
+}
+
+func GetAttr(element *html.Node, attrName string) (string, bool) {
+	for _, attr := range element.Attr {
+		if attr.Key == attrName {
+			return attr.Val, true
+		}
+	}
+	return "", false
+}
+
+func NewRequestFromForm(form *html.Node, currentURL *url.URL) (*http.Request, error) {
```

Comment:
I'm not normally a nit-picky kind of guy, but how about some doc about what the heck this doing.